### ISSUE TITLE
L_SEVERITY_NONE was not declared in this scope

### ISSUE THREAD

**Issue by webe3** [sirfz/tesserocr#38]
I am trying to do a pip3 and pip install of tesserocr on Debian stretch and getting the following error:

```
# pip3 install tesserocr
Collecting tesserocr
  Using cached tesserocr-2.1.3.tar.gz
Building wheels for collected packages: tesserocr
  Running setup.py bdist_wheel for tesserocr ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-3coz3mk5/tesserocr/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpmrdg6puopip-wheel- --python-tag cp35:
  running bdist_wheel
  running build
  running build_ext
  Supporting tesseract v3.04.01
  Configs from pkg-config: {'libraries': ['tesseract', 'lept'], 'cython_compile_time_env': {'TESSERACT_VERSION': 197633}, 'include_dirs': ['/usr/include']}
  cythoning tesserocr.pyx to tesserocr.cpp
  building 'tesserocr' extension
  creating build
  creating build/temp.linux-x86_64-3.5
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.5-MLq5fN/python3.5-3.5.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include -I/usr/include/python3.5m -c tesserocr.cpp -o build/temp.linux-x86_64-3.5/tesserocr.o
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  tesserocr.cpp: In function ‘PyObject* PyInit_tesserocr()’:
  tesserocr.cpp:24651:18: error: ‘L_SEVERITY_NONE’ was not declared in this scope
     setMsgSeverity(L_SEVERITY_NONE);
                    ^~~~~~~~~~~~~~~
  tesserocr.cpp:24651:33: error: ‘setMsgSeverity’ was not declared in this scope
     setMsgSeverity(L_SEVERITY_NONE);
                                   ^
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for tesserocr
  Running setup.py clean for tesserocr
Failed to build tesserocr
Installing collected packages: tesserocr
  Running setup.py install for tesserocr ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-3coz3mk5/tesserocr/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-j2y3y173-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    Supporting tesseract v3.04.01
    Configs from pkg-config: {'cython_compile_time_env': {'TESSERACT_VERSION': 197633}, 'include_dirs': ['/usr/include'], 'libraries': ['lept', 'tesseract']}
    skipping 'tesserocr.cpp' Cython extension (up-to-date)
    building 'tesserocr' extension
    creating build
    creating build/temp.linux-x86_64-3.5
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.5-MLq5fN/python3.5-3.5.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include -I/usr/include/python3.5m -c tesserocr.cpp -o build/temp.linux-x86_64-3.5/tesserocr.o
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    tesserocr.cpp: In function ‘PyObject* PyInit_tesserocr()’:
    tesserocr.cpp:24651:18: error: ‘L_SEVERITY_NONE’ was not declared in this scope
       setMsgSeverity(L_SEVERITY_NONE);
                      ^~~~~~~~~~~~~~~
    tesserocr.cpp:24651:33: error: ‘setMsgSeverity’ was not declared in this scope
       setMsgSeverity(L_SEVERITY_NONE);
                                     ^
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-3coz3mk5/tesserocr/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-j2y3y173-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-3coz3mk5/tesserocr/
```

**Comment by webe3** [comment]
L_SEVERITY_NONE is defined in an enum in /usr/include/leptonica/environ.h

enum {
    L_SEVERITY_EXTERNAL = 0,   /* Get the severity from the environment   */
    L_SEVERITY_ALL      = 1,   /* Lowest severity: print all messages     */
    L_SEVERITY_DEBUG    = 2,   /* Print debugging and higher messages     */
    L_SEVERITY_INFO     = 3,   /* Print informational and higher messages */
    L_SEVERITY_WARNING  = 4,   /* Print warning and higher messages       */
    L_SEVERITY_ERROR    = 5,   /* Print error and higher messages         */
    L_SEVERITY_NONE     = 6    /* Highest severity: print no messages     */
};

When I do a grep on /usr/include/leptonica/ environ.h is the only place it shows up:

# ls
allheaders.h  arrayaccess.h  bbuffer.h	  bmfdata.h  bmp.h     dewarp.h      environ.h	heap.h	   jbclass.h  list.h   pix.h   queue.h	 readbarcode.h	regutils.h  stringcode.h  watershed.h
alltypes.h    array.h	     bilateral.h  bmf.h      ccbord.h  endianness.h  gplot.h	imageio.h  leptwin.h  morph.h  ptra.h  rbtree.h  recog.h	stack.h     sudoku.h
root@brente:/usr/include/leptonica# grep L_SEVERITY_NONE *
environ.h: *  L_SEVERITY_NONE.                                                      *
environ.h:    L_SEVERITY_NONE     = 6    /* Highest severity: print no messages     */
environ.h:  #define MINIMUM_SEVERITY      L_SEVERITY_NONE    /*!< Compile-time default */
environ.h:  #define DEFAULT_SEVERITY      L_SEVERITY_NONE    /*!< Run-time default */

**Comment by sirfz** [comment]
It seems the installation cannot find the environ.h header file in your environment. Does `leptonica/allheaders.h` exist in `/usr/include`? What version of leptonica do you have installed?

**Comment by webe3** [comment]
On Tue, 2017-03-14 at 14:26 -0700, Fayez wrote:
> It seems the installation cannot find the environ.h header file in
> your environment. Does leptonica/allheaders.h exist in /usr/include?
> What version of leptonica do you have installed?
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Yes, allheaders.h is there and it includes environ.h.  I have version
1.74.1-1 of leptonica-dev from Debian stretch/testing.

**Comment by sirfz** [comment]
Can you please try installing the [tesseract4](https://github.com/sirfz/tesserocr/tree/tesseract4) version? Just clone the repo and run `pip install -U .` from its root directory.

Also make sure you have `pkg-config` installed (via apt) which makes it easier for the script to find the include and library dirs.

**Comment by webe3** [comment]
On Thu, 2017-03-16 at 10:35 -0700, Fayez wrote:
> Can you please try installing the tesseract4 version? Just clone the
> repo and run pip install -U . from its root directory.
> Also make sure you have pkg-config installed (via apt) which makes it
> easier for the script to find the include and library dirs.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

I think I found the problem.  I don't know how I got into the situation
but on my Debian stretch computer, I see
leptonica 1.69-1
leptonica-progs 1.74.1-1
leptonica-dev 1.74.1-1

I'm not sure where leptonica 1.69-1 came from but it put the includes
into /usr/local/include/leptonica.
leptonica-dev 1.74.1-1 put the includes into /usr/include/leptonica.
It looks like the pip install found the ones in
/usr/local/include/leptonica which is the older ones.  I removed the
directory /usr/local/include/leptonica and pip install worked.

Thanks
